### ISSUE TITLE
Updating to Python 3

### ISSUE THREAD

**Issue by tur-ium** [chocjy/randomized-LS-solvers#2]
# TLDR
I've updated the Python syntax to Python 3. I'm hitting some issues, which might be Spark related, or Numpy related. Posting to check if you have any ideas what might be happening?

# Environment
1. Python = Python 3.9
2. Spark + Scala + Java 

`spark-submit --version`

```
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 3.5.1
      /_/
                        
Using Scala version 2.12.18, OpenJDK 64-Bit Server VM, 11.0.22
Branch HEAD
Compiled by user heartsavior on 2024-02-15T11:24:58Z
Revision fd86f85e181fc2dc0f50a096855acf83a6cc5d9c
``` 
# Reproduction steps
1. Installed Spark and required Python packages
2. Updated syntax from Python 2 to Python 3. Please clone [the fork I've made](https://github.com/tur-ium/randomized-LS-solvers). I also added a `requirements.txt` for pip
3. `/bin/bash /home/artur/PycharmProjects/randomized-LS-solvers/test/run_test.sh`

# Findings
In summary there se
1.  TypeError: expected a sequence of integers or a single integer, got '1000.0'
2. TypeError: slice indices must be integers or None or have an __index__ method
4. TypeError: <lambda>() missing 1 required positional argument: 'pa' (I think this is coming from `projections.py`, I removed the unnecessary braces that were causing trouble in Python 3)
5.  TypeError
```
 in process
    self.PA[i] = np.dot(np.random.randn(self.c, r), data) / np.sqrt(self.c)
  File "numpy/random/mtrand.pyx", line 1287, in numpy.random.mtrand.RandomState.randn
  File "numpy/random/mtrand.pyx", line 1448, in numpy.random.mtrand.RandomState.standard_normal
  File "_common.pyx", line 636, in numpy.random._common.cont
TypeError: 'float' object cannot be interpreted as an integer
```

<details>
<summary>Full stack trace</summary>

[stack_trace.txt](https://github.com/chocjy/randomized-LS-solvers/files/14775784/stack_trace.txt)

</details>

# Disclaimer
I'm pretty fresh to Spark.
